File: src/main/java/infinitespire/rewards/VoidShardReward.java
package infinitespire.rewards;

import com.badlogic.gdx.graphics.Texture;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.UIStrings;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Reward;
import infinitespire.patches.RewardItemTypeEnumPatch;

public class VoidShardReward extends Reward {

	private static final Texture ICON = InfiniteSpire.Textures.getUITexture("topPanel/avhari/voidShard.png");
	private static final UIStrings STRINGS = CardCrawlGame.languagePack.getUIString("VoidShard");

	public int amountOfShards;

	public VoidShardReward(){
		this(1);
	}

	public VoidShardReward(int amount) {
		super(ICON, makeDescription(amount), RewardItemTypeEnumPatch.VOID_SHARD);
		this.amountOfShards = amount;
	}

	public static String makeDescription(int amount) {
		if(amount > 1) {
			return amount + STRINGS.TEXT[4];
		}
		return amount + STRINGS.TEXT[3];
	}

	@Override
	public boolean claimReward() {
		InfiniteSpire.gainVoidShards(amountOfShards);
		return true;
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/rewards/VoidShardReward.kt
package com.blanktheevil.infinitespire.rewards

import basemod.abstracts.CustomReward
import com.blanktheevil.infinitespire.extensions.addVoidShard
import com.blanktheevil.infinitespire.extensions.languagePack
import com.blanktheevil.infinitespire.patches.EnumPatches
import com.blanktheevil.infinitespire.textures.Textures

class VoidShardReward(
  val amountOfShards: Int = 1
) : CustomReward(ICON, getDesc(amountOfShards), EnumPatches.RewardType.VOID_SHARD) {
  companion object {
    private val ICON = Textures.ui.get("topPanel/avhari/voidShard.png")
    private val strings = languagePack.getUIString("VoidShard")

    private fun getDesc(value: Int): String {
      return if (value > 1)
        "$value${strings.TEXT[4]}"
      else
        "$value${strings.TEXT[3]}"
    }
  }

  override fun claimReward(): Boolean {
    addVoidShard(amountOfShards)
    return true
  }
}